omniscidb/QueryEngine/CostModel/IterativeCostModel.cpp

```diff
@@ -0,0 +1,40 @@
+#include "IterativeCostModel.h"
+#include "Dispatchers/ProportionBasedExecutionPolicy.h"
+
+#include <cmath>
+
+namespace costmodel {
+
+std::unique_ptr<policy::ExecutionPolicy> IterativeCostModel::predict(
+    QueryInfo queryInfo) {
+        std::lock_guard<std::mutex> l(latch_);
```

Comment:
I don't mind having a non-const `predict` method, in general, to account for a feedback loop in the future (even though it might be cleaner to have a separate update method), but the locking here looks way too pessimistic. Do we really need to synchronize here? If a critical section is unavoidable, can we at least narrow the scope?